org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BundleUtils.java

```diff
@@ -205,10 +208,32 @@ private static void installBundle(BundleContext context, Set<Bundle> bundlesToSt
 	 * @throws BundleException
 	 */
 	private static void uninstallBundle(Set<Bundle> bundlesToStart, Set<Bundle> toRefresh, Bundle bundle) throws BundleException {
-		bundle.uninstall();
-		JavaLanguageServerPlugin.logInfo("Uninstalled " + bundle.getLocation());
-		toRefresh.add(bundle);
-		bundlesToStart.remove(bundle);
+		if (checkBundle(bundle)) {
+			bundle.uninstall();
+			JavaLanguageServerPlugin.logInfo("Uninstalled " + bundle.getLocation());
+			toRefresh.add(bundle);
+			bundlesToStart.remove(bundle);
+		}
+	}
+
+	private static boolean checkBundle(Bundle bundle) throws BundleException {
+		FrameworkWiring frameworkWiring = getFrameworkWiring();
+		List<Bundle> list = new ArrayList<>();
+		list.add(bundle);
+		Collection<Bundle> bundles = frameworkWiring.getDependencyClosure(list);

```

Comment:
If the list doesn't need to be mutable, then:
frameworkWiring.getDependencyClosure(Collections.singletonList(bundle));

Reply:
Fixed.